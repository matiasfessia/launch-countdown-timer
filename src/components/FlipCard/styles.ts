import { css } from '@emotion/react';

// @todo move all these const to global const styling file
const background = '#231d2b';
const fontSize = 40;
const fontWeight = 700;
const outerFontColor = '#d35072';
const outerBackgroundColor = '#35354f';
const innerFontColor = '#d35072';
const innerBackgroundColor = '#2c2c44';

export const styles = {
  outerCard: css`
    position: relative;
    width: 70px;
    height: 70px;
    &:before {
      position: absolute;
      z-index: 0;
      overflow: hidden;
      width: 100%;
      height: 50%;
      background: ${outerBackgroundColor};
      box-shadow: 0px 5px 2px 0px #1a1a24;
      border-radius: 0 0 5px 5px;
      bottom: 2px;
      content: attr(data-value);
      color: ${outerFontColor};
      font-size: ${fontSize}px;
      font-weight: ${fontWeight};
      line-height: 0px;
      text-align: center;
    }
  `,
  innerCard: css`
    &:before {
      position: absolute;
      z-index: 1;
      overflow: hidden;
      width: 100%;
      height: 50%;
      background: ${innerBackgroundColor};
      border-radius: 5px 5px 0 0;
      border-bottom: 1px solid rgb(0 0 0 / 30%);
      background-clip: padding-box;
      content: attr(data-value);
      color: ${innerFontColor};
      font-size: ${fontSize}px;
      font-weight: ${fontWeight};
      text-align: center;
      line-height: 66px;
    }
  `,
  bite: css`
    position: absolute;
    z-index: 3;
    width: 3px;
    height: 5px;
    top: 33px;
    border-radius: 50%;
    background-color: ${background};
  `,
  leftBite: css`
    left: -1px;

    border-bottom-right-radius: 10px;
    border-top-right-radius: 10px;
  `,
  rightBite: css`
    right: -1px;
    border-bottom-left-radius: 10px;
    border-top-left-radius: 10px;
  `
};
