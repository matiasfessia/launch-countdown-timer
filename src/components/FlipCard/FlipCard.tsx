/** @jsxImportSource @emotion/react */
import { styles } from './styles';

export interface Props {
  value: string;
}

export const FlipCard = ({ value }: Props) => {
  return (
    <div css={styles.outerCard} data-value={value}>
      <div css={styles.innerCard} data-value={value} />
      <div css={[styles.bite, styles.leftBite]} />
      <div css={[styles.bite, styles.rightBite]} />
    </div>
  );
};

export default FlipCard;
