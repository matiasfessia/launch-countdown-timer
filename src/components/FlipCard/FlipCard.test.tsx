import { render } from '@testing-library/react';
import { FlipCard } from './FlipCard';

it('should renders correctly', () => {
  const { container } = render(<FlipCard title="title" value="33" />);
  expect(container).toMatchSnapshot();
});
