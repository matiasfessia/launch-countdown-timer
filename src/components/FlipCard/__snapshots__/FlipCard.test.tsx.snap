// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="css-1822xkh-FlipCard"
  >
    <div
      class="css-y5ro21-FlipCard"
      data-value="33"
    >
      <div
        class="css-nq5f8h-FlipCard"
        data-value="33"
      />
      <div
        class="css-xp1t8d-FlipCard"
      />
      <div
        class="css-18hlcob-FlipCard"
      />
    </div>
    <h2
      class="css-xzkfzk-FlipCard"
    />
  </div>
</div>
`;
